compiler_opt/rl/corpus.py

```diff
@@ -224,6 +224,40 @@ def __call__(self, k: int, n: int = 10) -> List[ModuleSpec]:
     return list(results)
 
 
+class WholeCorpusSampler(Sampler):
+  """Returns the entire corpus every time a sample is requested."""
+
+  # We need to include this method even if it is a useless super delegation
+  # as super().__init__ is an abstract method that we need to override to
+  # execute it without throwing an error.
+  def __init__(self, module_specs: Tuple[ModuleSpec]):  # pylint: disable=useless-super-delegation
+    super().__init__(module_specs)
+
+  def reset(self):
+    pass
+
+  def __call__(self, k: int, n: int = 10) -> List[ModuleSpec]:
+    """Returns the entire corpus a list of module specs.
+
+    Args:
+      k: The number of modules to sample. This must be equal to the number
```

Comment:
why do we even need "k" here?

Reply:
We have to include `k` and `n` or otherwise we get lint errors given the base class has those as parameters.

Reply:
another way to understand the problem, though, would be to say that given an initial module-level corpus, and a set of subsets of modules, you get a new corpus where a corpus element is such a subset of modules.

then "k" counts how many of those subsets.

Reply:
Sure, but I don't plan on supporting multiple entrypoints to begin with. That would also require more stuff passed around here as we would additionally need to pass the trace associated with that entrypoint associated with this corpus subset.